src/infinity/libraries/Bridges/LibCoorestCarbon.sol

```diff
@@ -0,0 +1,130 @@
+        uint256 retiredAmount
+    );
+
+    error FeePercentageGreaterThanDivider();
+    error FeeRetireDividerIsZero();
+
+    /**
+     * @notice Retires CCO2
+     * @dev Use this function to retire CCO2.
+     * @dev This function assumes that checks to carbonToken are make higher up in call stack.
+     * @dev It's important to know that Coorest transfers fee portion to it's account & rest amount is burned
+     * @param carbonToken CCO2 token address.
+     * @param retireAmount The amount of underlying tokens to retire.
+     * @return poccId POCC Certificate Id.
+     */
+    function retireCarbonToken(
+        address carbonToken,
+        uint256 retireAmount,
+        LibRetire.RetireDetails memory details
+    ) internal returns (uint256 poccId) {
+        require(details.beneficiaryAddress != address(0), "Beneficiary Address can't be 0");
+
+        IERC20(carbonToken).approve(C.coorestPool(), retireAmount);
+
+        // Transfer PoCC to beneficiary that's minted in favor RA
+        // when Coorest retires CCO2
+        poccId = ICoorest(C.coorestPool()).mintPOCC(
+            retireAmount,
+            details.retirementMessage,
+            // Coorest expects owner as a string as it's to added to pocc token
+            LibMeta.addressToString(details.beneficiaryAddress)
+        );
+
+        IERC721(C.coorestPoCCToken()).safeTransferFrom(address(this), details.beneficiaryAddress, poccId);
+
+        LibRetire.saveRetirementDetails(
+            carbonToken,
+            address(0),
+            retireAmount,
+            details.beneficiaryAddress,
+            details.beneficiaryString,
+            details.retirementMessage
+        );
+
+        emit CarbonRetired(
+            LibRetire.CarbonBridge.COOREST,
+            details.retiringAddress,
+            details.retiringEntityString,
+            details.beneficiaryAddress,
+            details.beneficiaryString,
+            details.retirementMessage,
+            carbonToken,
+            address(0),
+            retireAmount
+        );
+    }
+
+    /**
+     * @notice Calculates the Coorest fee that needs to be added to desired retire amount
+     * @dev Use this function to compute the Coorest fee.
+     * @dev This function assumes that checks to carbonToken are make higher up in call stack
+     * @param carbonToken     CCO2 token address
+     * @param amount          The amount of underlying tokens to retire.
+     * @return feeAmount      Fee charged by Coorest.
+     */
+    function getSpecificRetirementFee(address carbonToken, uint256 amount) external view returns (uint256 feeAmount) {
+        uint256 retireAmount = amount;
+        require(retireAmount > 0, "Retire amount should be greater than 0");
+
+        FeeParams memory feeParams = getFeePercent(carbonToken);
+
+        feeAmount =
+            ((retireAmount * feeParams.feeRetireDivider) / (feeParams.feeRetireDivider - feeParams.feeRetireBp)) -
+            retireAmount;
+    }
```

Comment:
in the cco2 contract the fee is calculated:
```solidity
     uint256 amountForCoorest = _amount * mintingPercentage / decimalRatio ;
```

the calculation here will result in a slightly larger feeAmount, correct? Will that cause a discrepancy between the retired amount emitted by the RA and the retired amount emitted by cco2?